recipes/configs/llama3/8B_qlora_single_device.yaml

```diff
@@ -23,6 +23,7 @@ model:
   apply_lora_to_output: False
   lora_rank: 8
   lora_alpha: 16
+  low_cpu_ram: False
```

Comment:
Given that we have to modify the state dict hook, I see why it makes sense to put this in the model config. But it does feel a bit weird to me since it's not really a property of the model (more just the model is a convenient place for us to know that we're gonna have to upcast NF4 tensors). 

I wonder if we can instead define a standalone config, parse it in the recipe with e.g. `low_cpu_ram = cfg.get("low_cpu_ram", False)`, then use that to overwrite the `reparametrize_as_dtype_state_dict_post_hook`. Maybe a bit hacky, but we can at least assert that the expected state dict hook is there before replacing it to ensure that we aren't adding this onto any old non-QLoRA model. Is that obviously worse? (Mainly I want to avoid our model classes having to know or care about low-level details of how they're gonna be checkpointed)